### ISSUE TITLE
TLS support in Spring auto configuration

### ISSUE THREAD

**Issue by gquintana** [line/armeria#1659]
Spring Boot Configuration 
```
server:
  port: 9411
  ssl:
    enabled: true
    key-alias: "host.name.com"
    key-store: "keystore.jks"
    key-store-password: "changeme"
    trust-store: "truststore.jks"
    trust-store-password: "changeme"
armeria:
  ports:
    - port: 9411
      protocol: HTTPS

```
Exception raised
```
Caused by: java.lang.IllegalArgumentException: TLS not configured; cannot serve HTTPS
        at com.linecorp.armeria.server.ServerBuilder.build(ServerBuilder.java:1340) ~[armeria-0.80.0.jar!/:?]
        at com.linecorp.armeria.spring.ArmeriaAutoConfiguration.armeriaServer(ArmeriaAutoConfiguration.java:113) ~[armeria-spring-boot-autoconfigure-0.80.0.jar!/:?]
        at com.linecorp.armeria.spring.ArmeriaAutoConfiguration$$EnhancerBySpringCGLIB$$2f3f223e.CGLIB$armeriaServer$0(<generated>) ~[armeria-spring-boot-autoconfigure-0.80.0.jar!/:?]
        at com.linecorp.armeria.spring.ArmeriaAutoConfiguration$$EnhancerBySpringCGLIB$$2f3f223e$$FastClassBySpringCGLIB$$b07f737b.invoke(<generated>) ~[armeria-spring-boot-autoconfigure-0.80.0.jar!/:?]
```
There is a TODO https://github.com/line/armeria/blob/armeria-0.80.0/spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaSettings.java#L43

Version: 0.80.0

**Comment by trustin** [comment]
Are you using a non-reactive stack?

**Comment by adriancole** [comment]
looks like this is related to an issue in zipkin raised by the same user https://github.com/openzipkin/zipkin/issues/2449

**Comment by minwoox** [comment]
Closed by #1663.

**Comment by gquintana** [comment]
Works great, thanks @imasahiro
